doc/Iothub_sdk_options.md

```diff
@@ -1,119 +1,178 @@
+
+
+<a name="IotHub_options"></a>
+
+## IoT Hub Device and Module Client Options
+You can use the options below for IoT Hub connections.  These options are declared in [iothub_client_options.h][iothub-client-options-h].  
 
-<a name="IotHub_option"></a>
+You may also use [common transport options](#general_options) options.
+
+**NOTE: Uploading files to Azure (e.g. IoTHubDeviceClient_UploadToBlobAsync) does not support this complete set of options.  See [here](#upload-options) for more.**
 
-### IoThub_Client
 
 | Option Name                       | Option Define                   | Value Type         | Description
 |-----------------------------------|---------------------------------|--------------------|-------------------------------
-| `"messageTimeout"`              | OPTION_MESSAGE_TIMEOUT         | tickcounter_ms_t*  | (DEPRECATED) Timeout used for message on the message queue
-| `"blob_upload_timeout_secs"`  | OPTION_BLOB_UPLOAD_TIMEOUT_SECS | size_t*            | Timeout in seconds of blob uploads
-| `"product_info"`                | OPTION_PRODUCT_INFO             | const char*        | User defined Product identifier sent to the IoThub service
-| `"TrustedCerts"`                | OPTION_TRUSTED_CERT             | const char*        | Azure Server certificate used to validate TLS connection to iothub
-| `"retry_interval_sec"`          | OPTION_RETRY_INTERVAL_SEC       |  int*              | Amount of seconds between retries when using the interval retry policy
+| `"logtrace"`                      | OPTION_LOG_TRACE                | bool*              | Turn on and off log tracing for the transport
+| `"messageTimeout"`                | OPTION_MESSAGE_TIMEOUT          | tickcounter_ms_t*  | (DEPRECATED) Timeout used for message on the message queue
+| `"product_info"`                  | OPTION_PRODUCT_INFO             | const char*        | User defined Product identifier sent to the IoT Hub service
+| `"retry_interval_sec"`            | OPTION_RETRY_INTERVAL_SEC       | unsigned int*      | Number of seconds between retries when using the interval retry policy.  (Not supported for HTTP transport.)
+| `"retry_max_delay_secs"`          | OPTION_RETRY_MAX_DELAY_SECS     | unsigned int*      | Maximum number of seconds a retry delay when using linear backoff, exponential backoff, or exponential backoff with jitter policy.  (Not supported for HTTP transport.)
+| `"sas_token_lifetime"`            | OPTION_SAS_TOKEN_LIFETIME       | size_t*            | Length of time in seconds used for lifetime of SAS token.
+| `"do_work_freq_ms"`               | OPTION_DO_WORK_FREQUENCY_IN_MS  | unsigned int *     | Specifies how frequently the worker thread spun by the convenience layer will wake up, in milliseconds.  The default is 1 millisecond.  The maximum allowable value is 100.  (Convenience layer APIs only)
```

Comment:
It appears the type is `tickcounter_ms_t` https://github.com/Azure/azure-iot-sdk-c/issues/1011#issuecomment-494508855. Kindly ignore if otherwise.

Reply:
Good catch.  I will fix this in next round.  I'll double check the provisioning client equivalent of this and fixup here if in correct.  Thanks

Reply:
OK - fixed locally, put a link to header that has `tickcounter_ms_t` as this isn't a standard data type, and fixed up prov client (which takes a uint16*).  Will push if/when more feedback comes &certainly before merge.